File: sdks/csharp/Windjammer.SDK/Time.cs
namespace Windjammer.SDK;

/// <summary>
/// Time information for the current frame.
/// </summary>
public class Time
{
    /// <summary>Time since last frame in seconds.</summary>
    public float DeltaSeconds { get; set; } = 0.016f; // ~60 FPS
    
    /// <summary>Total time since application start in seconds.</summary>
    public float TotalSeconds { get; set; } = 0.0f;
    
    /// <summary>Current frame number.</summary>
    public int FrameCount { get; set; } = 0;

    /// <inheritdoc/>
    public override string ToString() => $"Time(delta={DeltaSeconds}, total={TotalSeconds})";
}

File: sdks/javascript/src/time.ts
/**
 * Time utilities.
 */

/**
 * Time information for the current frame.
 */
export class Time {
  /** Time since last frame in seconds */
  deltaSeconds: number = 0.016; // ~60 FPS
  
  /** Total time since application start in seconds */
  totalSeconds: number = 0.0;
  
  /** Current frame number */
  frameCount: number = 0;

  toString(): string {
    return `Time(delta=${this.deltaSeconds}, total=${this.totalSeconds})`;
  }
}

